### PR TITLE
CURA-4555: Fixing "Open with"

### PR DESCRIPTION
**Pull request by thopiekar**

### CODE REVIEW

**Comment by thopiekar** [comment]
Discussion about the issue: https://github.com/Ultimaker/cura-build/commit/0bf1f72a4d555a4570a72fae0669b685df4e8cff

File: cmake/NSIS.template.in [hunk 1]
```diff
@@ -869,7 +869,8 @@ Section "Uninstall"
   ${UnRegisterExtension} ".x3d" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.x3d.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
 
   ;Remove executable from MUIcache
-  DeleteRegKey HKCR "Software\Microsoft\Windows\Shell\MuiCache\$INSTDIR\Cura.exe.FriendlyAppName"
+  DeleteRegKey HKCR "Local Settings\Software\Microsoft\Windows\Shell\MuiCache\$INSTDIR\Cura.exe.ApplicationCompany"
+  DeleteRegKey HKCR "Local Settings\Software\Microsoft\Windows\Shell\MuiCache\$INSTDIR\Cura.exe.FriendlyAppName"
```
Comment (ChrisTerBeke) [inline cmake/NSIS.template.in]: @alekseisasin might this fix your issue from this morning?

Comment (thopiekar) [reply cmake/NSIS.template.in]: @ChrisTerBeke He was testing my branch here.. I think so.
However, I'm building a new installer and will give a try here.

Comment (thopiekar) [reply cmake/NSIS.template.in]: What I notice now, too, is that I wrote "MUIcache" wrong 👀

**Comment by thopiekar** [comment]
The new installer is on my NAS now (See CURA ticket for URL).
To make the MuiCache fix take effect it needs an (uninstall + install)². Otherwise you don't use the new changes inside the uninstall section.

**Comment by thopiekar** [comment]
@alekseisasin , do you know for what "Hash" is for?
image

**Comment by thopiekar** [comment]
Ok, crossing fingers and hoping we don't need to regenerate the hash for Windows..

**Comment by thopiekar** [comment]
Ah! The hash needs to be updated. Otherwise Microsoft is showing us this one: 🖕

**Comment by thopiekar** [comment]
v4 of the installer is now on my NAS.
As long as we can't generate the hash inside NSIS, we are stuck here.
I think having "Slice with Ultimaker Cura" is enough for now 😉

**Comment by alekseisasin** [comment]
I will try to find the solution, but first I want to be sure that after each try catch I have default story in regedit, either will get wrong testing result.

**Comment by LipuFei** [comment]
@thopiekar You branch fails on the build server with the following errors:
```
Section: "Open 3MF files with Cura"
!insertmacro: RegisterExtensionCall
Error: labels must not begin with 0-9, -, :, or a space.
Error in macro RegisterExtension_ on macroline 50
Error in macro CallArtificialFunction on macroline 11
Error in macro RegisterExtensionCall on macroline 7
```

**Comment by thopiekar** [comment]
Hmm, let me see whether I can reproduce it and fix it, if it happens over here.

**Comment by thopiekar** [comment]
Hmm, get the same error. Give me some time to look over it.

**Comment by thopiekar** [comment]
Before:
snip_20171208125101

After:
image

**Comment by thopiekar** [comment]
@LipuFei Can you retry now?

**Comment by LipuFei** [comment]
It's not working for me :(
image

**Comment by thopiekar** [comment]
I'm sorry to hear that. All your changes to make it working, are in "test_open_cura", right?
Can see whether I will find time to take a look onto that.
Btw, is 3MF the only one, which does not work? When I was working on these fixes, I only took attention on STL files and expected that it will work the same for all the other formats.

Regards!

**Comment by LipuFei** [comment]
@thopiekar That was a different attempt. I tried that one and yours `CURA-4555-thopiekars-experiments`, neither worked.

**Comment by alekseisasin** [comment]
@thopiekar I tried "CURA-4555-thopiekars-experiments branch" and could apply Cura for .stl or .3mf as default application.
I created a virtual disk with a clean windows 7 and installed cura. After it Cura is  still not selected as default application, but if I open **Open With** I could easily select Cura for .stl and .3mf file extensions

Maybe our local windows are so messed with different installations and it is hard to test each time, to do this we need a clean windows installation. 

And as I see it works for "Open with", but not set as **default application** after installation.

**Comment by thopiekar** [comment]
I'm glad to hear that it works like intended.
At the moment I don't have the time to hack around and see how to reset the current association to the default, which we are setting with these changes during installation.

It would be nice, if we could at least merge this bundle of changes before the next release and can care about a solution for resetting the association in a future release. So we have it at least "half-fixed" at the moment and the customers can't say we don't care at all.

I will have more time for this in at least 5 weeks. Of course you can try to find a solution for this and I'm curious what the results for a solution will be, but currently I'm going to be too busy to look into any issues and PRs for many weeks (worst case).

**Pull request merged**